The stickmen from the paper robe will now actually attack other players

ATH1909 opened the pull request:
## About The Pull Request

This PR removes the "neutral" faction from the stickmen made by the paper robe. This feature can be toggled if you just want your stickmen to mostly just vibe. It also slightly tweaks what you say when you summon a stickman with a paper robe.

The stickmen created by the paper robe have a maxHealth of 50 now (instead of 100). "Wild" stickmen and the stickmen summoned by Mjor the Creative aren't affected by this change.

The paper robe now has a cooldown between summons of 5 seconds (up from 3 seconds).

## Why It's Good For The Game

Basically, before this PR, stickmen would just kinda hang around and not do much, even if a traitor just shot their creator in the face. Now, they'll attack (on sight) most things that aren't their creator or a stickman that their creator has made (or something else that shares their creator's non-neutral faction(s)), so you can sic them on other players.

I basically just copied the code for this from dehydrated carp plushies, so any bugs with those will probably carry over to the paper robe's minions.

I halved the maxHealth of these things and increased the cooldown time of the robe because shitting out a 100 maxHealth minion every 3 seconds sounds like it'd be actually cancerous to fight against.

## Changelog
:cl: ATHATH
balance: The stickmen created by the paper robe will now attack players (but not their creator, of course) on their own. This feature can be toggled using the new "Toggle Stick Minion Aggressiveness" button. In other words, the paper robe isn't a glorified paperweight anymore.
balance: The stickmen created by the paper robe have a maxHealth of 50 now (instead of 100). "Wild" stickmen and the stickmen summoned by Mjor the Creative aren't affected by this change.
balance: The paper robe now has a cooldown between summons of 5 seconds (up from 3 seconds).
/:cl:

ATH1909 committed:
get to work you lazy stickmen

ATH1909 commented:
Wait, holy shit, these things have 100 health, lemme halve that (for summoned ones only) real quick.

ATH1909 committed:
no spamming 100 health mobs everywhere

ATH1909 commented:
bam, done

Shadowflame909 commented:
arent these the weak ones without guns.

100 health makes sense doc. Especially since you gotta get shot like 30 times to get them.

necromanceranne commented:
No it doesn't make sense. The robes have a cooldown of two seconds. You literally can produce an infinite quantity of bodies and no gun short of assault rifles are going to adequately get past the huge mob of dudes. I would highly recommend increasing the cooldown to a more reasonable amount if you do intend to make it actually attack players.

The health isn't the issue. It's the mob spam. Also I am fairly sure it spawns more than one stickman at a time. Been a while since I got the robes.

Tlaltecuhtli commented:
cant you make it a toggle? the only time i saw them used was to cannon fodder xenos (they attack xenos )

Tharcoonvagh commented:
Fixes #49134

ExcessiveUseOfCobblestone commented:
this should be a toggleable similar to the hiero staff

ATH1909 commented:
> No it doesn't make sense. The robes have a cooldown of two seconds. You literally can produce an infinite quantity of bodies and no gun short of assault rifles are going to adequately get past the huge mob of dudes. I would highly recommend increasing the cooldown to a more reasonable amount if you do intend to make it actually attack players.
> 
> The health isn't the issue. It's the mob spam. Also I am fairly sure it spawns more than one stickman at a time. Been a while since I got the robes.

The cooldown is actually 3 seconds, and it spawns 1 stickman per "casting".

ATH1909 commented:
> Fixes #49134

Er, not quite. This doesn't affect "natural" stickmen or the stickmen from Mjor, ONLY the stickmen from the paper robe.

ATH1909 commented:
> this should be a toggleable similar to the hiero staff

That sounds like a good idea, I'll look into that.

ATH1909 committed:
adds a toggle for stick minion aggressiveness

also increases the cooldown time of the paper robe and removes a S L E E P

ATH1909 commented:
Alright, I've added a "Toggle Stick Minion Aggressiveness" button, removed a sleep() from the code of the paper robe, and increased the paper robe's cooldown time between summons to 5 seconds (up from 3 seconds).

Rohesie commented:
How about a max number of stickmen active at any time?

optimumtact commented on code/modules/clothing/suits/wiz_robe.dm:
```diff
@@ -146,38 +146,44 @@
+	var/mob/living/M = new /mob/living/simple_animal/hostile/stickman/paperrobe(get_turf(usr)) //these guys have half the maximum health of a normal stickman, to make them less cancerous to fight against when spammed
+	var/list/factions = usr.faction.Copy()
+	if(angry_sticks)
+		for(var/F in factions)
```
you could just do `factions.remove("neutral")`

ATH1909 replied:
thanks

ExcessiveUseOfCobblestone commented on code/modules/clothing/suits/wiz_robe.dm:
```diff
@@ -146,38 +146,44 @@
-	sleep(30)
-	src.robe_charge = TRUE
-	to_chat(usr, "<span class='notice'>\The robe hums, its internal magic supply restored.</span>")
+	timer = world.time + 50 //5 second cooldown between stickman spawns
```
time define

ATH1909 committed:
web edit

web edit

ATH1909 committed:
Update wiz_robe.dm

ATH1909 committed:
"thanks", oranges

ATH1909 commented:
Alright, I undid oranges's suggestion (because it causes the code to not compile) and went back to how dehydrated carp remove the neutral faction from the carp they spawn.

ExcessiveUseOfCobblestone commented on code/modules/clothing/suits/wiz_robe.dm:
```diff
@@ -146,38 +146,44 @@
+	var/mob/living/M = new /mob/living/simple_animal/hostile/stickman/paperrobe(get_turf(usr)) //these guys have half the maximum health of a normal stickman, to make them less cancerous to fight against when spam-summoned
+	var/list/factions = usr.faction.Copy()
+	if(angry_sticks)
+		for(var/F in factions)
+			if(F == "neutral")
+				factions -= F  //this is so that they'll attack other players and NPCs and stuff, but still won't attack the person who made them or each other
```
just do `factions -= neutral`, if it doesnt exist it shouldnt freak out.

ATH1909 replied:
are you absolutely, 100% sure about this?

ExcessiveUseOfCobblestone replied:
it is just a list.

Rohesie replied:
`LAZYREMOVE(factions, "neutral")` should work alright
If the element is not on the list nothing will happen.

Qustinnus commented:
if you're going to change the balance on a mob it should be in the title.

ATH1909 commented:
> if you're going to change the balance on a mob it should be in the title.

it's in the changelog

and again, it ONLY APPLIES TO THE STICKMEN FROM THE ROBE. THE ONES IN THE SPACE RUIN (and the ones summoned by Mjor) ARE NOT AFFECTED.

Rohesie commented:
Are you still working on this?
Have you addressed the balance concern of infinite stickmen?

kriskog commented:
merge conflict

Rohesie commented:
Are you still planning on working on this?

ATH1909 committed:
Merge branch 'master' into steeck

Rohesie commented:
Still no max cap of active stick figures at any given time?

ExcessiveUseOfCobblestone commented:
You have generated new atom-bound variables/procs without proper documentation.

Please click [here ](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)for a guide on how to properly document these items.

ExcessiveUseOfCobblestone commented on code/modules/clothing/suits/wiz_robe.dm:
```diff
@@ -1,231 +1,235 @@
+	desc = "A robe held together by various bits of clear-tape and paste."
+	icon_state = "wizard-paper"
+	inhand_icon_state = "wizard-paper"
+	var/timer = 0 //when we'll next be able to use the summon stickman ability
+	var/angry_sticks = TRUE // have you set the robe to make stickmen who WON'T have the neutral faction? note that this WON'T affect stickmen that (whom?) you've already made
```
new variables should be below inherited

Cyberboss reviewed (changes requested):
Cobby's review + docs

Rohesie commented:
If you re-add it, please add a cap to how many of these can be alive at the same time.
